c-dependencies/js-compute-runtime/host_call.h

```diff
@@ -1,56 +1,104 @@
 #ifndef JS_COMPUTE_RUNTIME_HOST_CALL_H
 #define JS_COMPUTE_RUNTIME_HOST_CALL_H
 
-#include "js-compute-builtins.h"
+// TODO: remove these once the warnings are fixed
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Winvalid-offsetof"
+#include "jsapi.h"
+#pragma clang diagnostic pop
 
 #include "xqd.h"
 
+typedef enum class FastlyStatus {
+  // Success value.
+  // This indicates that a hostcall finished successfully.
+  Ok = 0,
+  // Generic error value.
+  // This means that some unexpected error occurred during a hostcall.
+  Error = 1,
+  // Invalid argument.
+  Inval = 2,
+  // Invalid handle.
+  // Thrown when a handle is not valid. E.G. No dictionary exists with the given name.
+  BadF = 3,
+  // Buffer length error.
+  // Thrown when a buffer is too long.
+  BufLen = 4,
+  // Unsupported operation error.
+  // This error is thrown when some operation cannot be performed, because it is not supported.
+  Unsupported = 5,
+  // Alignment error.
+  // This is thrown when a pointer does not point to a properly aligned slice of memory.
+  BadAlign = 6,
+  // Invalid HTTP error.
+  // This can be thrown when a method, URI, header, or status is not valid. This can also
+  // be thrown if a message head is too large.
+  HttpInvalid = 7,
+  // HTTP user error.
+  // This is thrown in cases where user code caused an HTTP error. For example, attempt to send
+  // a 1xx response code, or a request with a non-absolute URI. This can also be caused by
+  // an unexpected header: both `content-length` and `transfer-encoding`, for example.
+  HttpUser = 8,
+  // HTTP incomplete message error.
+  // This can be thrown when a stream ended unexpectedly.
+  HttpIncomplete = 9,
+  // A `None` error.
+  // This status code is used to indicate when an optional value did not exist, as opposed to
+  // an empty value.
+  None = 10,
+  // Message head too large.
+  HttpHeadTooLarge = 11,
+  // Invalid HTTP status.
+  HttpInvalidStatus = 12,
+} FastlyStatus;
```

Comment:
Removing the other half of the `typedef` :)
```suggestion
};
```